custom/icds/models.py

```diff
@@ -0,0 +1,148 @@
+
+from casexml.apps.case.models import CommCareCase
+from corehq.apps.userreports.expressions.getters import transform_date
+
+
+from .constants import *
+
+
+class ChildHealthCaseRow(object):
+    def __init__(self, case, snapshot_start_date, snapshot_end_date, forms_provider=[]):
+        """
+        Snapshot of the case in the duration specified by snapshot_start_date and snapshot_end_date
+
+        args:
+            case: case of type 'child_health'
+            snapshot_start_date: Start date of duration to snapshot the case on
+            snapshot_end_date: End date of duration to snapshot the case on.
+
+        dates can be date/datetime objects or a valid strings
+        """
+        if not isinstance(case, CommCareCase):
+            case = CommCareCase.wrap(case)
```

Comment:
> If possible it would be best to leave the case as JSON.

agree with this. if this isn't possible we should minimize the number of places we make the couch assumption and abstract them out so we can plug in sql adapters later.